src/main/java/com/firebolt/jdbc/exception/ServerError.java

```diff
@@ -126,12 +136,12 @@ public Error(String code, String name, Severity severity, Source source, String
             this.location = location;
         }
 
-        Error(JSONObject json) {
+        Error(JSONObject json, boolean isError) {
             this(json.optString("code", null),
                     json.optString("name", null),
                     json.optEnum(Severity.class, "severity"),
                     ofNullable(json.optString("source", null)).map(Source::fromText).orElse(Source.UNKNOWN),
-                    json.optString("description", null),
+                    json.optString(isError ? "description" : "detail", null),
```

Comment:
We can probably eliminate the need for `isError` altogether by using something like `json.optString("description", json.optString("detail", null))`